{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>class</th>\n",
       "      <th>abstract</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>B</td>\n",
       "      <td>the 4 202 353 bp genome of the alkaliphilic ba...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>A</td>\n",
       "      <td>the complete 1751377-bp sequence of the genome...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>E</td>\n",
       "      <td>in 1992 we started assembling an ordered libra...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>E</td>\n",
       "      <td>the aim of this study is to measure human mito...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>B</td>\n",
       "      <td>the amino acid sequence of the spirulina maxim...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>6</td>\n",
       "      <td>B</td>\n",
       "      <td>the genus xanthomonas is a diverse and economi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>7</td>\n",
       "      <td>B</td>\n",
       "      <td>the complete nucleotide sequence of the genome...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>8</td>\n",
       "      <td>B</td>\n",
       "      <td>the complete genome sequence of caulobacter cr...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>9</td>\n",
       "      <td>V</td>\n",
       "      <td>the complete dna sequence of the a2 strain of ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>10</td>\n",
       "      <td>B</td>\n",
       "      <td>the complete genomic sequence of corynebacteri...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   id class                                           abstract\n",
       "0   1     B  the 4 202 353 bp genome of the alkaliphilic ba...\n",
       "1   2     A  the complete 1751377-bp sequence of the genome...\n",
       "2   3     E  in 1992 we started assembling an ordered libra...\n",
       "3   4     E  the aim of this study is to measure human mito...\n",
       "4   5     B  the amino acid sequence of the spirulina maxim...\n",
       "5   6     B  the genus xanthomonas is a diverse and economi...\n",
       "6   7     B  the complete nucleotide sequence of the genome...\n",
       "7   8     B  the complete genome sequence of caulobacter cr...\n",
       "8   9     V  the complete dna sequence of the a2 strain of ...\n",
       "9  10     B  the complete genomic sequence of corynebacteri..."
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "text_df = pd.read_csv(\"./trg.csv\")\n",
    "\n",
    "#define constant\n",
    "labelAConstant = \"zclass_A\"\n",
    "labelBConstant = \"zclass_B\"\n",
    "labelEConstant = \"zclass_E\"\n",
    "labelVConstant = \"zclass_V\"\n",
    "\n",
    "# Display the first 10 rows\n",
    "text_df.head(10)\n",
    "\n",
    "\n",
    "# A simple boolean representation of word appearances\n",
    "#import numpy as np \n",
    "\n",
    "# Find all the words\n",
    "#words = []\n",
    "#for i in range(0, text_df.shape[0]):\n",
    "#for i in range(0, 1):    \n",
    "#    abstract = text_df[\"abstract\"][i]\n",
    "#    words_abstract = abstract.split(\" \") # Split when there is a space\n",
    "#    words.extend(words_abstract)\n",
    "\n",
    "    \n",
    "# Remove duplicate words and sort them alphabetically\n",
    "#words = list(np.unique(np.sort(words)))\n",
    "#words = list(np.unique(words))\n",
    "#words = list(np.sort(np.unique(words)))\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "from collections import Counter\n",
    "import numpy as np\n",
    "\n",
    "\"\"\"\n",
    "Retrieve worlds from abstract. Manipulate words if required\n",
    "\"\"\"\n",
    "def retrieveWordListFromAbstract(abstract):\n",
    "    redudancyWords = [\"IS\", \"A\", \"THE\", \"OF\", \"AND\", \"IN\", \"TO\"]\n",
    "    words = [x.strip().upper() for x in abstract.split(\" \")]\n",
    "    words = [ x for x in words if x != \"\"] \n",
    "    words = [ x for x in words if not(x in redudancyWords)] \n",
    "    \n",
    "    return words\n",
    "\n",
    "\"\"\"\n",
    "Create default dictionary data for featureList\n",
    "return a dictionary with key is feature and value is 0\n",
    "\"\"\" \n",
    "def createDefaultDictionaryFromFeatureList(featureList):\n",
    "    result = {}\n",
    "    for feature in featureList:\n",
    "        result[feature] = 0\n",
    "    return result    \n",
    "\n",
    "\"\"\"\n",
    "Map abstractData to an dictionary which has key is feature \n",
    "and value = 1 if that feature appear in abstractData, otherwise value = 0\n",
    "\"\"\"\n",
    "#transform each abstract data to attribute list\n",
    "def transformData(abstractData, featureList):\n",
    "    words_abstract = retrieveWordListFromAbstract(abstractData)\n",
    "    unique_words = np.unique(words_abstract)\n",
    "    \n",
    "    #create default value for dictionary list\n",
    "    dictionaryResult = createDefaultDictionaryFromFeatureList(featureList)\n",
    "    \n",
    "    # assign the feature which appear in abstract data\n",
    "    for index in range(0, len(unique_words)):\n",
    "        word = unique_words[index]\n",
    "        foundInList = word in featureList\n",
    "        if foundInList:\n",
    "            dictionaryResult[word] = 1            \n",
    "    \n",
    "    return dictionaryResult;\n",
    "\n",
    "\"\"\"\n",
    "Create featureList by composing the 1000 most frequency keywords in abstractDataList\n",
    "\"\"\"\n",
    "def get1000MostFrequencyFeatureFrom(abstractDataList):\n",
    "    words = []\n",
    "    for index in range(0, abstractDataList.size):\n",
    "        abstract = abstractDataList[index]\n",
    "        words_abstract = retrieveWordListFromAbstract(abstract)\n",
    "        words.extend(words_abstract)\n",
    "    occurenceList = Counter(words)\n",
    "    \n",
    "    #get 1000 most common objects with their occurences\n",
    "    #most_1000_frequency = occurenceList.most_common(1000)\n",
    "    most_1000_frequency = occurenceList.most_common(100)\n",
    "    \n",
    "    features = np.empty(len(most_1000_frequency), dtype=object)\n",
    "    #features = []\n",
    "    # withraw all the keywords\n",
    "    for index in range(0, len(most_1000_frequency)):\n",
    "        features[index] = most_1000_frequency[index][0]\n",
    "        #features.append(most_1000_frequency[index][0])\n",
    "    \n",
    "    #return features\n",
    "    return features.tolist()\n",
    "\n",
    "\"\"\"\n",
    "Create featureList by composing all words which appear in abstractList\n",
    "\"\"\"\n",
    "def retrieveFeatureFrom(abstractDataList):\n",
    "    words = []\n",
    "    for index in range(0, abstractDataList.size):\n",
    "        abstract = abstractDataList[index]\n",
    "        words_abstract = retrieveWordListFromAbstract(abstract)\n",
    "        words.extend(words_abstract)\n",
    "\n",
    "    words = list(np.sort(np.unique(words)))\n",
    "    \n",
    "    return words\n",
    "\n",
    "\n",
    "# Create dataset for training from featureList and abstractList\n",
    "def buildTrainingDataset(featureList, abstractList):\n",
    "    dataFrame = {}\n",
    "    aLength = len(abstractList)\n",
    "    fLength = len(featureList)\n",
    "    \n",
    "    #create default list\n",
    "    for featureIndex in range(0, fLength):\n",
    "        featureName = featureList[featureIndex]\n",
    "        dataFrame[featureName] = []\n",
    "        #dataFrame[featureName]= np.empty(aLength, dtype=object)\n",
    "        \n",
    "    for index in range(0, aLength):\n",
    "        abstract = abstractList[index]\n",
    "        dictionaryForAbstract = transformData(abstract, featureList)\n",
    "        \n",
    "        for typleFeatureItem in dictionaryForAbstract.items():\n",
    "            featureName = typleFeatureItem[0]\n",
    "            featureValue = typleFeatureItem[1]\n",
    "            \n",
    "            dataFrame[featureName].append(featureValue)\n",
    "            \n",
    "    return dataFrame\n",
    "\n",
    "\n",
    "def buildTestingDataset(featureList, abstractList):\n",
    "    dataFrame = {}\n",
    "    aLength = len(abstractList)\n",
    "    fLength = len(featureList)\n",
    "    \n",
    "    #create default list\n",
    "    for featureIndex in range(0, fLength):\n",
    "        featureName = featureList[featureIndex]\n",
    "        dataFrame[featureName] = []\n",
    "        #dataFrame[featureName]= np.empty(aLength, dtype=object)\n",
    "        \n",
    "    for index in range(0, aLength):\n",
    "        abstract = abstractList[index]\n",
    "        dictionaryForAbstract = transformData(abstract, featureList)\n",
    "        \n",
    "        for typleFeatureItem in dictionaryForAbstract.items():\n",
    "            featureName = typleFeatureItem[0]\n",
    "            featureValue = typleFeatureItem[1]\n",
    "            \n",
    "            dataFrame[featureName].append(featureValue)\n",
    "            \n",
    "    return dataFrame\n",
    "\n",
    "def buildClass(classList):\n",
    "    uniqueClassList = np.unique(classList)\n",
    "    noClass = len(classList)\n",
    "    resultValue = {}\n",
    "    \n",
    "    #assign default value\n",
    "    for classValue in uniqueClassList:\n",
    "        tempList = np.zeros(noClass, dtype = int)\n",
    "        resultValue[\"zclass_\"+classValue] = tempList.tolist()\n",
    "        \n",
    "    #assign the real value for resultValue\n",
    "    for index in range(0, noClass):\n",
    "        classValue = classList[index]\n",
    "        resultValue[\"zclass_\" + classValue][index] = 1\n",
    "    return resultValue\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "def countFeatureAndClassByValue(featureName, featureValue, className, classValue, featureDF, classDF):\n",
    "    nRows = featureDF.shape[1]\n",
    "    count = 0\n",
    "    for index in range(0, nRows):\n",
    "        fValue = featureDF[featureName][index]\n",
    "        cValue = classDF[className][index]\n",
    "        if fValue == featureValue and cValue == classValue:\n",
    "            count = count + 1\n",
    "    return count\n",
    "\n",
    "\n",
    "# Find the probability of featureName = featureValue given className = 1 with featuresDF and classDF\n",
    "# featureValue = 1 or 0\n",
    "def calculateProbabilityOf(featureName, featureValue, className, classValue, trainDF):\n",
    "    noClass = np.count_nonzero(trainDF[className])\n",
    "    condition = (trainDF[featureName] == featureValue) & (trainDF[className] == classValue)\n",
    "    count = trainDF[condition].shape[0]\n",
    "    \n",
    "    return count/noClass;\n",
    "    \n",
    "\n",
    "def retrieveClassHasValueAtIndex(index, classDF):\n",
    "    rows = classDF.iloc[index]\n",
    "    columnNames = classDF.columns.values\n",
    "    for className in columnNames:\n",
    "        if rows.at[className] == 1:\n",
    "            return className\n",
    "    return 0\n",
    "    \n",
    "# train\n",
    "def train(featuresDF, classDF):\n",
    "    #calculate probability of each class\n",
    "    nTotal = featuresDF.shape[0]\n",
    "\n",
    "    # calculate\n",
    "    columnsNameList = classDF.columns.values\n",
    "    \n",
    "    \n",
    "    #calculate indexList\n",
    "    featureList = featuresDF.columns.values\n",
    "    indexList = []\n",
    "    for index in range(0, len(featureList)):\n",
    "        featureIsZero = featureList[index] + \"=0\"\n",
    "        featureIsOne = featureList[index] + \"=1\"\n",
    "        indexList.append(featureIsZero)\n",
    "        indexList.append(featureIsOne)\n",
    "    indexList.append(\"total\")\n",
    "    \n",
    "\n",
    "    #compose the target model\n",
    "    result = np.zeros((len(indexList), len(columnsNameList)), dtype = int)\n",
    "    resultDF = pd.DataFrame(result , columns = columnsNameList, index=indexList)\n",
    "    \n",
    "    for rowIndex in range(0, nTotal):\n",
    "        #calculate value for feature \n",
    "        for featureName in featureList:\n",
    "            featureValue = featuresDF[featureName][rowIndex]\n",
    "            rowName = featureName + \"=\" + str(featureValue)\n",
    "            activeClassAtIndex = retrieveClassHasValueAtIndex(rowIndex, classDF)\n",
    "            resultDF[activeClassAtIndex][rowName] += 1 \n",
    "            \n",
    "    #calculate total for class\n",
    "    for className in columnsNameList:\n",
    "        count = training_df[(training_df[className] == 1)].shape[0]\n",
    "        resultDF[className][\"total\"] = count\n",
    "    \n",
    "    return resultDF\n",
    "\n",
    "def test(testDF):\n",
    "    return 0\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>zclass_A</th>\n",
       "      <th>zclass_B</th>\n",
       "      <th>zclass_E</th>\n",
       "      <th>zclass_V</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HOMOLOGY=0</th>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HOMOLOGY=1</th>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HERE=0</th>\n",
       "      <td>2</td>\n",
       "      <td>12</td>\n",
       "      <td>10</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HERE=1</th>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>USING=0</th>\n",
       "      <td>1</td>\n",
       "      <td>14</td>\n",
       "      <td>10</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>USING=1</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>total</th>\n",
       "      <td>2</td>\n",
       "      <td>14</td>\n",
       "      <td>12</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            zclass_A  zclass_B  zclass_E  zclass_V\n",
       "HOMOLOGY=0         1        12        12         1\n",
       "HOMOLOGY=1         1         2         0         0\n",
       "HERE=0             2        12        10         1\n",
       "HERE=1             0         2         2         0\n",
       "USING=0            1        14        10         1\n",
       "USING=1            1         0         2         0\n",
       "total              2        14        12         1"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "training_df = pd.read_csv(\"./data/test_data.csv\", )\n",
    "training_df.shape\n",
    "\n",
    "cross_validation_training_feature_df = training_df.drop([labelAConstant, labelBConstant, labelEConstant, labelVConstant], axis=1)\n",
    "cross_validation_traing_class_df = training_df[[labelAConstant, labelBConstant, labelEConstant, labelVConstant]]\n",
    "#cross_validation_test_df[labelAConstant]\n",
    "#model = train(training_df, [labelAConstant, labelBConstant, labelEConstant, labelVConstant])\n",
    "#countFeatureAndClassByValue(\"treatment\", 1, \"zclass_A\", 1, cross_validation_training_feature_df, cross_validation_traing_class_df)\n",
    "#condition = (training_df[\"treatment\"] == 1) & (training_df[\"zclass_A\"] == 1)\n",
    "#value = training_df[condition]\n",
    "#value.shape\n",
    "\n",
    "#classM = retrieveClassHasValueAtIndex(1, cross_validation_traing_class_df)\n",
    "#classM\n",
    "\n",
    "model = train(cross_validation_training_feature_df, cross_validation_traing_class_df)\n",
    "model\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 507,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "filter_df\n",
      "    HOMOLOGY  HERE  USING  zclass_A  zclass_B  zclass_E  zclass_V\n",
      "0          0     0      0         0         1         0         0\n",
      "4          0     0      0         0         1         0         0\n",
      "5          0     1      0         0         1         0         0\n",
      "6          0     0      0         0         1         0         0\n",
      "7          0     0      0         0         1         0         0\n",
      "9          0     0      0         0         1         0         0\n",
      "11         0     0      0         0         1         0         0\n",
      "12         0     0      0         0         1         0         0\n",
      "14         1     0      0         0         1         0         0\n",
      "15         0     1      0         0         1         0         0\n",
      "16         1     0      0         0         1         0         0\n",
      "23         0     0      0         0         1         0         0\n",
      "25         0     0      0         0         1         0         0\n",
      "26         0     0      0         0         1         0         0\n",
      "shape\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(14, 7)"
      ]
     },
     "execution_count": 507,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#filterElm = (training_df[\"zclass_A\"] == 1) & (training_df[\"HERE\"] == 0)\n",
    "filterElm = (training_df[\"zclass_B\"] == 1)\n",
    "filter_df = training_df[filterElm]\n",
    "print(\"filter_df\")\n",
    "print(filter_df)\n",
    "print(\"shape\")\n",
    "filter_df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 334,
   "metadata": {},
   "outputs": [],
   "source": [
    "#test = buildClass(text_df[\"class\"])\n",
    "#tf_test = pd.DataFrame(test)\n",
    "#tf_test.to_csv(\"./class_value.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 471,
   "metadata": {},
   "outputs": [],
   "source": [
    "#featureTest = [\"a1\", \"a2\", \"a3\"]\n",
    "#defaultValues = createDefaultDictionaryFromFeatureList(featureTest)\n",
    "#defaultValues\n",
    "\n",
    "featureList = get1000MostFrequencyFeatureFrom(text_df[\"abstract\"])\n",
    "#featureList = retrieveFeatureFrom(text_df[\"abstract\"])\n",
    "#print(\"featureList\", featureList)\n",
    "\n",
    "featureDataSet = buildTrainingDataset(featureList, text_df[\"abstract\"])\n",
    "classDataSet = buildClass(text_df[\"class\"])\n",
    "trainingDataSet = {**featureDataSet, **classDataSet}\n",
    "df = pd.DataFrame(trainingDataSet)\n",
    "\n",
    "testData = pd.read_csv(\"./tst.csv\")\n",
    "testDataSet = buildTestingDataset(featureList, testData[\"abstract\"])\n",
    "tf = pd.DataFrame(testDataSet)\n",
    "\n",
    "df.to_csv(\"./data/training_100_feature_data.csv\", index=False)\n",
    "tf.to_csv(\"./data/test_100_feature_data.csv\", index=False)\n",
    "\n",
    "#df\n",
    "#df.to_csv(\"./training_data.csv\", index=False)\n",
    "#print(\"test\", test)\n",
    "\n",
    "#transform data for abstract data\n",
    "#dictionaryList = transformData(text_df[\"abstract\"][0], featureList)\n",
    "#dictionaryList\n",
    "#df = pd.DataFrame(dictionaryList)\n",
    "\n",
    "#test = retrieveFeatureFrom(text_df[\"abstract\"])\n",
    "#len(test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 469,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0       0\n",
       "1       1\n",
       "2       0\n",
       "3       0\n",
       "4       0\n",
       "       ..\n",
       "3995    0\n",
       "3996    0\n",
       "3997    0\n",
       "3998    0\n",
       "3999    0\n",
       "Name: zclass_A, Length: 4000, dtype: int64"
      ]
     },
     "execution_count": 469,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#training_df = pd.read_csv(\"./training_data.csv\")\n",
    "#training_df[0]\n",
    "#df[\"zclass_B\"]\n",
    "df[\"zclass_A\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "metadata": {},
   "outputs": [
    {
     "ename": "SyntaxError",
     "evalue": "invalid syntax (<ipython-input-167-6683d84fb750>, line 1)",
     "output_type": "error",
     "traceback": [
      "\u001b[0;36m  File \u001b[0;32m\"<ipython-input-167-6683d84fb750>\"\u001b[0;36m, line \u001b[0;32m1\u001b[0m\n\u001b[0;31m    training_df[,0]\u001b[0m\n\u001b[0m                ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m invalid syntax\n"
     ]
    }
   ],
   "source": [
    "training_df[,0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "test[0] 1\n",
      "test [[1 0 0]\n",
      " [0 0 0]]\n"
     ]
    }
   ],
   "source": [
    "import numpy as np\n",
    "test = np.zeros((2,3), dtype = int)\n",
    "#features = np.empty(len(most_1000_frequency), dtype=string)\n",
    "#a = np.array(42)\n",
    "#b = np.array([1, 2, 3, 4, 5])\n",
    "#c = np.array([[1, 2, 3], [4, 5, 6]])\n",
    "#d = np.array([[[1, 2, 3], [4, 5, 6]], [[1, 2, 3], [4, 5, 6]]])\n",
    "\n",
    "#print(len(a))\n",
    "#print(b.ndim)\n",
    "#print(c.ndim)\n",
    "#print(d.ndim)\n",
    "test[0][0] = 1\n",
    "print(\"test[0]\", test[0][0])\n",
    "print(\"test\" , test)\n",
    "\n",
    "a = np.zeros(10)\n",
    "type(a)\n",
    "\n",
    "import numpy as np\n",
    "np.zeros((2,3), dtype = int)\n",
    "np.empty(10, dtype= str)\n",
    "\n",
    "\n",
    "fruits = ['apple', 'banana', 'cherry']\n",
    "\n",
    "x = \"cherry\" in fruits\n",
    "if x:\n",
    "    print(\"found\")\n",
    "else:\n",
    "    print(\"not found\")\n",
    "type(x)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "numpy.ndarray"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['', '', '', '', '', '', '', '', '', ''], dtype='<U1')"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "found\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "bool"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "dict_items([('brand', 'Ford'), ('model', 'Mustang'), ('year', 1964)])\n"
     ]
    }
   ],
   "source": [
    "car = {\n",
    "  \"brand\": \"Ford\",\n",
    "  \"model\": \"Mustang\",\n",
    "  \"year\": 1964\n",
    "}\n",
    "\n",
    "x = car.items()\n",
    "\n",
    "print(x)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "text_df.to_csv(\"/Users/mac/Desktop/NZ/UOA/COMPSCI_361/assignments/assignment_05/test.csv\")\n",
    "\n",
    "type(text_df[\"abstract\"])\n",
    "from collections import Counter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['blah', 'lots', 'of', 'spaces', 'here']"
      ]
     },
     "execution_count": 134,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "my_string = \"blah, lots  ,  of ,  spaces, here \"\n",
    "result = [x.strip() for x in my_string.split(',')]\n",
    "result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 341,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[\"'\",\n",
       " \"'87\",\n",
       " \"'a\",\n",
       " \"'a'\",\n",
       " \"'addiction'\",\n",
       " \"'ancient'\",\n",
       " \"'b'\",\n",
       " \"'bird\",\n",
       " \"'birds\",\n",
       " \"'c3-type'\",\n",
       " \"'captive'\",\n",
       " \"'chaperonins'\",\n",
       " \"'classical'\",\n",
       " \"'core'\",\n",
       " \"'cysteine-rich'\",\n",
       " \"'deadh'\",\n",
       " \"'destabilization\",\n",
       " \"'dinosaur\",\n",
       " \"'ets\",\n",
       " \"'feral'\",\n",
       " \"'heat-shock'\",\n",
       " \"'hitch-hiking'\",\n",
       " \"'housekeeping'\",\n",
       " \"'immune'\",\n",
       " \"'inflammatory\",\n",
       " \"'insert'\",\n",
       " \"'inserted'\",\n",
       " \"'king\",\n",
       " \"'lcp30'\",\n",
       " \"'lego\",\n",
       " \"'lightning-strike\",\n",
       " \"'mdr-like'\",\n",
       " \"'motor\",\n",
       " \"'mr\",\n",
       " \"'neutralized'\",\n",
       " \"'occluding\",\n",
       " \"'open\",\n",
       " \"'open'\",\n",
       " \"'plasticity\",\n",
       " \"'processing'\",\n",
       " \"'pstaire'\",\n",
       " \"'shotgun'\",\n",
       " \"'silent'\",\n",
       " \"'tail'\",\n",
       " \"'tissue-specific'\",\n",
       " \"'transcriptional\",\n",
       " \"'ttp'\",\n",
       " \"'uptake'\",\n",
       " \"'wolf'\",\n",
       " '-',\n",
       " '--',\n",
       " '-1',\n",
       " '-10',\n",
       " '-1055',\n",
       " '-11',\n",
       " '-116',\n",
       " '-124',\n",
       " '-126',\n",
       " '-130',\n",
       " '-13548',\n",
       " '-1363',\n",
       " '-13859',\n",
       " '-14',\n",
       " '-142',\n",
       " '-1421',\n",
       " '-147',\n",
       " '-149',\n",
       " '-14q31',\n",
       " '-15',\n",
       " '-1561',\n",
       " '-16',\n",
       " '-17',\n",
       " '-18',\n",
       " '-180',\n",
       " '-19',\n",
       " '-1a2',\n",
       " '-1a2-like',\n",
       " '-2',\n",
       " '-20',\n",
       " '-200',\n",
       " '-21',\n",
       " '-210',\n",
       " '-213',\n",
       " '-215',\n",
       " '-219',\n",
       " '-220',\n",
       " '-24',\n",
       " '-24-12',\n",
       " '-24th',\n",
       " '-25',\n",
       " '-268',\n",
       " '-28',\n",
       " '-2a',\n",
       " '-3',\n",
       " '-30',\n",
       " '-32',\n",
       " '-33',\n",
       " '-330',\n",
       " '-3368',\n",
       " '-337',\n",
       " '-34',\n",
       " '-35',\n",
       " '-35-promoter',\n",
       " '-364',\n",
       " '-367',\n",
       " '-370',\n",
       " '-380',\n",
       " '-39',\n",
       " '-40',\n",
       " '-41',\n",
       " '-44',\n",
       " '-49',\n",
       " '-493',\n",
       " '-5',\n",
       " '-50',\n",
       " '-53',\n",
       " '-56',\n",
       " '-6',\n",
       " '-60',\n",
       " '-65',\n",
       " '-688',\n",
       " '-70',\n",
       " '-72',\n",
       " '-75',\n",
       " '-776',\n",
       " '-8',\n",
       " '-805',\n",
       " '-81',\n",
       " '-87',\n",
       " '-88',\n",
       " '-9',\n",
       " '-90',\n",
       " '-a25',\n",
       " '-ala-ala-pro-alaval-',\n",
       " '-arg-val-gln-gly-val-cys-phe-arg-met-tyr-thr-glu-asp-glu-ala-arg-lys-ile',\n",
       " '-asn-his-ser-thr-his-lys-ala-val-lys-cys-gly-asp-cys-his-his-pro-val-asn-gly-lys-glu-asn-tyr-gln-lys-cys-ala-thr-ala-gly-cys-his-asp-asn-met-asp-lys-lys-asp-lys-ser-ala-lys-gly-tyr-tyr-his-ala-met-his-asp-lys-gly-thr-lys-phe-lys-ser-cys-val-gly-cys-his-leu-glu-thr-ala-gly-ala-asp-ala-ala-lys-lys-lys-glu-leu-thr-gly-cys-lys-gly-ser-lys-cys-his-ser',\n",
       " '-asn-lys-thr-',\n",
       " '-associated',\n",
       " '-b',\n",
       " '-b37',\n",
       " '-b51',\n",
       " '-b57',\n",
       " '-b7',\n",
       " '-beta',\n",
       " '-c',\n",
       " '-cx2cx22cx2c-',\n",
       " '-dmyasfgl-nh2',\n",
       " '-fold',\n",
       " '-gly-val-val-gly-trp-val-lys-asn-thr-ser-lys-gly-thr-val-thr-gly-gln',\n",
       " '-golgi',\n",
       " '-gpyasfgl-nh2',\n",
       " '-gqyasfgl-nh2',\n",
       " '-hnel',\n",
       " '-htel',\n",
       " '-hvel',\n",
       " '-ii',\n",
       " '-ile-gly-ser-pro-ser-ser-arg-ile-asp-arg-thr-asn-phe-ser-asn-glu-lys-',\n",
       " '-independent',\n",
       " '-met-ile-arg-phe-nh2',\n",
       " '-phe-met-arg-phe-nh2',\n",
       " '-pro-pro-pro-nh2',\n",
       " '-pyafgl-nh2',\n",
       " '-sla',\n",
       " '-ssa',\n",
       " '-tyrosyl',\n",
       " '-v',\n",
       " '-val-gln-gly-pro-glu-glu-lys-val-asn-ser-met-lys-ser-trp-leu-ser-lys',\n",
       " '-x',\n",
       " '-yxfgl-nh2',\n",
       " '-zeaxanthin-beta-',\n",
       " '0',\n",
       " '0-100',\n",
       " '0-24',\n",
       " '0-31',\n",
       " '0-4',\n",
       " '00-15',\n",
       " '00-30',\n",
       " '00-45',\n",
       " '00-79',\n",
       " '000',\n",
       " '000-fold',\n",
       " '0001',\n",
       " '00017',\n",
       " '0003',\n",
       " '0004',\n",
       " '0005',\n",
       " '001',\n",
       " '0021',\n",
       " '003',\n",
       " '0035',\n",
       " '004',\n",
       " '005',\n",
       " '005-5',\n",
       " '007',\n",
       " '008',\n",
       " '01',\n",
       " '01-1',\n",
       " '01-22',\n",
       " '012',\n",
       " '013',\n",
       " '014',\n",
       " '015',\n",
       " '0163',\n",
       " '0169',\n",
       " '0188',\n",
       " '02',\n",
       " '02-88',\n",
       " '020',\n",
       " '0200',\n",
       " '0206',\n",
       " '021',\n",
       " '0237',\n",
       " '025',\n",
       " '025-8',\n",
       " '026',\n",
       " '0264',\n",
       " '02a',\n",
       " '03',\n",
       " '03-kb',\n",
       " '030',\n",
       " '031',\n",
       " '0314',\n",
       " '032',\n",
       " '035',\n",
       " '0354',\n",
       " '038-041',\n",
       " '04',\n",
       " '042',\n",
       " '044',\n",
       " '0449',\n",
       " '0455',\n",
       " '046',\n",
       " '047',\n",
       " '05',\n",
       " '05-09',\n",
       " '05-kilobase',\n",
       " '050-kb',\n",
       " '053',\n",
       " '0530',\n",
       " '054',\n",
       " '055',\n",
       " '06-kb',\n",
       " '06-kilobase-pair',\n",
       " '060',\n",
       " '062',\n",
       " '065',\n",
       " '066',\n",
       " '0660',\n",
       " '0669',\n",
       " '0678',\n",
       " '0688',\n",
       " '069',\n",
       " '07',\n",
       " '07-12',\n",
       " '07-4g',\n",
       " '07-kb',\n",
       " '07-kbp',\n",
       " '07-kilobase-long',\n",
       " '075',\n",
       " '08',\n",
       " '08-11',\n",
       " '08-15-kb-long',\n",
       " '08-kb',\n",
       " '08-kilobase',\n",
       " '080',\n",
       " '081',\n",
       " '084',\n",
       " '087',\n",
       " '08x10-4',\n",
       " '09',\n",
       " '09-0198',\n",
       " '09-kilobase',\n",
       " '090',\n",
       " '095',\n",
       " '1',\n",
       " '1-',\n",
       " '1-10',\n",
       " '1-13',\n",
       " '1-2',\n",
       " '1-20',\n",
       " '1-21',\n",
       " '1-240',\n",
       " '1-253',\n",
       " '1-27',\n",
       " '1-3',\n",
       " '1-3-beta-glucanase',\n",
       " '1-3-beta-glucanases',\n",
       " '1-4',\n",
       " '1-4-linked',\n",
       " '1-4n-acetylgalactosaminyltransferase',\n",
       " '1-6',\n",
       " '1-60',\n",
       " '1-65',\n",
       " '1-8',\n",
       " '1-9',\n",
       " '1-96',\n",
       " '1-aminocyclopropane-1-carboxylate',\n",
       " '1-aminocyclopropane-1-carboxylic',\n",
       " '1-antiproteinase',\n",
       " '1-antiproteinases',\n",
       " '1-antitrypsin',\n",
       " '1-at',\n",
       " '1-bp',\n",
       " '1-cad',\n",
       " '1-cads',\n",
       " '1-carboxy-3-chloro-45-dihydroxycyclohexa-26-diene',\n",
       " '1-cell',\n",
       " '1-d-old',\n",
       " '1-deoxy-xylulose-5-phosphate',\n",
       " '1-helix',\n",
       " '1-hour-dehydrated',\n",
       " '1-kb',\n",
       " '1-kilobase',\n",
       " '1-leaving',\n",
       " '1-myr-old',\n",
       " '1-nitropropane',\n",
       " '1-phosphate',\n",
       " '1-pi',\n",
       " '1-protease',\n",
       " '1-proteinase',\n",
       " '1-related',\n",
       " '1-rrna',\n",
       " '1-stearoyl-2-arachidonoyl',\n",
       " '1-subunit',\n",
       " '1-subunits',\n",
       " '10',\n",
       " '10-',\n",
       " '10-10',\n",
       " '10-12',\n",
       " '10-13',\n",
       " '10-15-fold',\n",
       " '10-2',\n",
       " '10-20',\n",
       " '10-20-fold',\n",
       " '10-24',\n",
       " '10-3',\n",
       " '10-4',\n",
       " '10-59',\n",
       " '10-6',\n",
       " '10-7',\n",
       " '10-8site',\n",
       " '10-9',\n",
       " '10-day',\n",
       " '10-fold',\n",
       " '10-kb',\n",
       " '10-kda',\n",
       " '10-kilobase',\n",
       " '10-min',\n",
       " '10-nm',\n",
       " '10-residue',\n",
       " '10-wk-old',\n",
       " '100',\n",
       " '100-',\n",
       " '100-115',\n",
       " '100-140',\n",
       " '100-base',\n",
       " '100-bp',\n",
       " '100-fold',\n",
       " '100-kda',\n",
       " '100-megabase',\n",
       " '1000',\n",
       " '1000-',\n",
       " '1000-fold',\n",
       " '1000-residue',\n",
       " '10000',\n",
       " '10000-fold',\n",
       " '100000',\n",
       " '1002',\n",
       " '100314',\n",
       " '1003450',\n",
       " '1004-amino',\n",
       " '10058',\n",
       " '1009',\n",
       " '100k',\n",
       " '101',\n",
       " '101-104',\n",
       " '101-amino',\n",
       " '101-kb',\n",
       " '1010',\n",
       " '1011',\n",
       " '1013',\n",
       " '1014',\n",
       " '1016',\n",
       " '1017',\n",
       " '101965',\n",
       " '102',\n",
       " '102-kda',\n",
       " '102-nt',\n",
       " '1020-bp',\n",
       " '1021',\n",
       " '1025',\n",
       " '1026-bp',\n",
       " '1027',\n",
       " '10277-10281',\n",
       " '1029',\n",
       " '10297',\n",
       " '103',\n",
       " '103-',\n",
       " '103-115',\n",
       " '103-base',\n",
       " '103-kda',\n",
       " '1030',\n",
       " '1032',\n",
       " '1033',\n",
       " '1034',\n",
       " '1035',\n",
       " '1037',\n",
       " '1038',\n",
       " '104',\n",
       " '104-215-day-old',\n",
       " '104-amino-acid',\n",
       " '1041',\n",
       " '1042519-base',\n",
       " '1043',\n",
       " '1043-1051',\n",
       " '10466',\n",
       " '105',\n",
       " '105-13',\n",
       " '105-kda',\n",
       " '105-kilobase',\n",
       " '1050',\n",
       " '10500',\n",
       " '10519',\n",
       " '1053',\n",
       " '10531',\n",
       " '1054',\n",
       " '10567',\n",
       " '1059',\n",
       " '105bp',\n",
       " '106',\n",
       " '1061-bp',\n",
       " '1065',\n",
       " '106516-bp',\n",
       " '10660',\n",
       " '10667-10672',\n",
       " '1068',\n",
       " '107',\n",
       " '1072314',\n",
       " '1074-1078',\n",
       " '1076',\n",
       " '1077',\n",
       " '108',\n",
       " '1080',\n",
       " '1080-bp',\n",
       " '108000',\n",
       " '1081',\n",
       " '1083-base-pair',\n",
       " '1085',\n",
       " '1086-base',\n",
       " '10897',\n",
       " '109',\n",
       " '109-kd',\n",
       " '109-kda',\n",
       " '10937',\n",
       " '1095',\n",
       " '1097',\n",
       " '1098',\n",
       " '10k',\n",
       " '10p12',\n",
       " '10q21',\n",
       " '10q22',\n",
       " '10q241-10q243',\n",
       " '11',\n",
       " '11-12',\n",
       " '11-13',\n",
       " '11-15',\n",
       " '11-16',\n",
       " '11-base',\n",
       " '11-bp',\n",
       " '11-containing',\n",
       " '11-dehydrocorticosterone',\n",
       " '11-fold',\n",
       " '11-fold-higher',\n",
       " '11-hydroxysteroid',\n",
       " '11-kb',\n",
       " '11-mb',\n",
       " '11-residue',\n",
       " '110',\n",
       " '110-kda',\n",
       " '110-kilodalton',\n",
       " '1100',\n",
       " '1100-bp',\n",
       " '11000',\n",
       " '11000-40000',\n",
       " '110000',\n",
       " '11039',\n",
       " '1104',\n",
       " '11059',\n",
       " '1107',\n",
       " '11089',\n",
       " '110917-193643',\n",
       " '111',\n",
       " '111-128',\n",
       " '111-base',\n",
       " '1110',\n",
       " '1111',\n",
       " '1111523',\n",
       " '11116',\n",
       " '11121',\n",
       " '1113',\n",
       " '111398',\n",
       " '111402-nucleotide',\n",
       " '111416',\n",
       " '11166130',\n",
       " '1119',\n",
       " '11195',\n",
       " '112',\n",
       " '112-kb',\n",
       " '1120',\n",
       " '112264',\n",
       " '11238',\n",
       " '1124',\n",
       " '1125',\n",
       " '112500',\n",
       " '1126-bp',\n",
       " '112635',\n",
       " '113',\n",
       " '113-120',\n",
       " '1131',\n",
       " '1131-nt',\n",
       " '11339-11344',\n",
       " '1138006',\n",
       " '114',\n",
       " '114-bp',\n",
       " '1140',\n",
       " '114000',\n",
       " '114012',\n",
       " '114047',\n",
       " '1141567',\n",
       " '114173',\n",
       " '1142',\n",
       " '1143',\n",
       " '1146',\n",
       " '1149910',\n",
       " '115',\n",
       " '115-120',\n",
       " '115-65',\n",
       " '115-amino-acid',\n",
       " '115-kb',\n",
       " '1150',\n",
       " '1151',\n",
       " '1151309-1318',\n",
       " '1157',\n",
       " '1158',\n",
       " '1159',\n",
       " '11592',\n",
       " '116',\n",
       " '116-kda',\n",
       " '116-kilodalton',\n",
       " '116000',\n",
       " '1161',\n",
       " '1164',\n",
       " '11665',\n",
       " '1168',\n",
       " '117',\n",
       " '117-bp',\n",
       " '117403',\n",
       " '1177787',\n",
       " '117a',\n",
       " '118',\n",
       " '118068',\n",
       " '1182',\n",
       " '118360',\n",
       " '1184',\n",
       " '1188',\n",
       " '119',\n",
       " '119-222',\n",
       " '119-bp',\n",
       " '1191',\n",
       " '1194',\n",
       " '1197',\n",
       " '119707',\n",
       " '11999-1018',\n",
       " '11a',\n",
       " '11bp',\n",
       " '11k',\n",
       " '11p14-151',\n",
       " '11p151',\n",
       " '11p155',\n",
       " '11q13',\n",
       " '11q13-q14',\n",
       " '11q131-q132',\n",
       " '11q143',\n",
       " '11q22-q23',\n",
       " '11q23',\n",
       " '11q25',\n",
       " '11th',\n",
       " '12',\n",
       " '12-15',\n",
       " '12-19',\n",
       " '12-24',\n",
       " '12-base-long',\n",
       " '12-bp',\n",
       " '12-dioxygenase',\n",
       " '12-dioxygenases',\n",
       " '12-fold',\n",
       " '12-kb',\n",
       " '12-kda',\n",
       " '12-o-tetradecanoylphorbol-13-acetate',\n",
       " '12-tetradecanoate',\n",
       " '12-transmembrane-domain',\n",
       " '120',\n",
       " '120-megabase',\n",
       " '1200',\n",
       " '12000',\n",
       " '120000',\n",
       " '120041',\n",
       " '1201',\n",
       " '1202',\n",
       " '1203',\n",
       " '12032',\n",
       " '1206-base-pair',\n",
       " '1207',\n",
       " '12080',\n",
       " '121',\n",
       " '121-124',\n",
       " '121-amino-acid',\n",
       " '12112',\n",
       " '1212',\n",
       " '1213',\n",
       " '12132-dalton',\n",
       " '1215',\n",
       " '1215176-182',\n",
       " '121524',\n",
       " '1217',\n",
       " '1218',\n",
       " '122',\n",
       " '122-205',\n",
       " '122-kda',\n",
       " '1222',\n",
       " '12228',\n",
       " '1223',\n",
       " '1223-bp',\n",
       " '1226',\n",
       " '1229-1240',\n",
       " '1229-1249',\n",
       " '123',\n",
       " '1230',\n",
       " '1230230-nt',\n",
       " '1233-bp',\n",
       " '12368',\n",
       " '1237',\n",
       " '1238',\n",
       " '1239-1247',\n",
       " '124',\n",
       " '124-131',\n",
       " '124-239',\n",
       " '1240',\n",
       " '124381',\n",
       " '1244',\n",
       " '1248',\n",
       " '124884',\n",
       " '1249',\n",
       " '125',\n",
       " '125-kb',\n",
       " '125-kilobase',\n",
       " '1250',\n",
       " '1250-nucleotide',\n",
       " '12500',\n",
       " '1252-base',\n",
       " '1257',\n",
       " '1258-base',\n",
       " '1259',\n",
       " '125i-anp',\n",
       " '125i-cgrp',\n",
       " '125i-cnp',\n",
       " '125i-labeled',\n",
       " '125i-npy',\n",
       " '125oh2d',\n",
       " '126',\n",
       " '126-residue',\n",
       " '1260-amino-acid',\n",
       " '1261',\n",
       " '126183-kda',\n",
       " '1264',\n",
       " '12642-12649',\n",
       " '1265',\n",
       " '1266-bp',\n",
       " '1267',\n",
       " '1268-bp',\n",
       " '1268755-nucleotide',\n",
       " '126k',\n",
       " '127',\n",
       " '127-132',\n",
       " '12708',\n",
       " '127107-115',\n",
       " '1277-bp',\n",
       " '128',\n",
       " '128-kb',\n",
       " '1281',\n",
       " '1281-bp',\n",
       " '1282',\n",
       " '12822',\n",
       " '1285',\n",
       " '129',\n",
       " '129-140',\n",
       " '129-nucleotide',\n",
       " '129-residue',\n",
       " '1290',\n",
       " '129000-536000',\n",
       " '1296',\n",
       " '1297',\n",
       " '129sv',\n",
       " '12a',\n",
       " '12alpha-hydroxylase',\n",
       " '12c2',\n",
       " '12p',\n",
       " '12p1123-p121',\n",
       " '12q12',\n",
       " '12q13',\n",
       " '12q23-q24',\n",
       " '12s',\n",
       " '12s-16s',\n",
       " '13',\n",
       " '13-10',\n",
       " '13-34',\n",
       " '13-aa',\n",
       " '13-acetate',\n",
       " '13-amino-acid',\n",
       " '13-base-pair',\n",
       " '13-beta-glucanase',\n",
       " '13-beta-glucanases',\n",
       " '13-bp',\n",
       " '13-diaminopropane',\n",
       " '13-dipropyl-8-4-acrylatephenylxanthine',\n",
       " '13-dipropyl-8-sulfophenylxanthine',\n",
       " '13-fold',\n",
       " '13-kb',\n",
       " '13-kbp',\n",
       " '13-kda',\n",
       " '13-kilobase-pair',\n",
       " '13-subunit',\n",
       " '130',\n",
       " '130-140',\n",
       " '130-fold',\n",
       " '130-kda',\n",
       " '1300',\n",
       " '13000',\n",
       " '130000',\n",
       " '1302',\n",
       " '13032',\n",
       " '1305',\n",
       " '130963',\n",
       " '131',\n",
       " '131-amino',\n",
       " '131-kb',\n",
       " '1311',\n",
       " '13134',\n",
       " '131573--592',\n",
       " '1317-base-pair',\n",
       " '13173',\n",
       " '132',\n",
       " '132-amino-acid',\n",
       " '132-bp',\n",
       " '1323',\n",
       " '13235',\n",
       " '1324',\n",
       " '1326',\n",
       " '133',\n",
       " '133000',\n",
       " '1331-bp',\n",
       " '1332-1343',\n",
       " '1332-bp',\n",
       " '133200',\n",
       " '13347-13354',\n",
       " '13350',\n",
       " '1338',\n",
       " '13384-13388',\n",
       " '134',\n",
       " '1341',\n",
       " '1341-nucleotide',\n",
       " '13421',\n",
       " '1343',\n",
       " '1344',\n",
       " '1345-bp',\n",
       " '1347',\n",
       " '13480',\n",
       " '135',\n",
       " '135-amino-acid',\n",
       " '135-fold',\n",
       " '135-kb',\n",
       " '135-kilobase',\n",
       " '1350',\n",
       " '13500',\n",
       " '1351',\n",
       " '1354226-nt',\n",
       " '136',\n",
       " '136-kb',\n",
       " '1360',\n",
       " '13600',\n",
       " '1365',\n",
       " '1367185',\n",
       " '137',\n",
       " '1374',\n",
       " '1375',\n",
       " '1377',\n",
       " '1378',\n",
       " '1378-nucleotide',\n",
       " '13794',\n",
       " '138',\n",
       " '138-140',\n",
       " '1380',\n",
       " '1387',\n",
       " '1389',\n",
       " '139',\n",
       " '139-144',\n",
       " '1393',\n",
       " '1395',\n",
       " '13978-13985',\n",
       " '1399',\n",
       " '13998',\n",
       " '13c',\n",
       " '13k',\n",
       " '13q12-q13',\n",
       " '13q33-34',\n",
       " '14',\n",
       " '14-',\n",
       " '14-17',\n",
       " '14-25',\n",
       " '14-3-3',\n",
       " '14-40',\n",
       " '14-5',\n",
       " '14-amino-acid',\n",
       " '14-diamino-2-butanone',\n",
       " '14-fold',\n",
       " '14-kb',\n",
       " '14-kda',\n",
       " '14-kilobase',\n",
       " '14-member',\n",
       " '14-n-acetylgalactosaminyltransferase',\n",
       " '140',\n",
       " '140-bp',\n",
       " '1400',\n",
       " '14000',\n",
       " '14000-fold',\n",
       " '14000-molecular-weight',\n",
       " '140000',\n",
       " '1401',\n",
       " '14028-1s',\n",
       " '1404',\n",
       " '14073',\n",
       " '14094',\n",
       " '141',\n",
       " '14105',\n",
       " '1411',\n",
       " '14119',\n",
       " '1412-4',\n",
       " '1413',\n",
       " '14130',\n",
       " '14131',\n",
       " '142',\n",
       " '142006',\n",
       " '14213',\n",
       " '1425',\n",
       " '1426',\n",
       " '14284',\n",
       " '143',\n",
       " '143-amino-acid',\n",
       " '143000',\n",
       " '1431',\n",
       " '143161',\n",
       " '14384-14389',\n",
       " '1439',\n",
       " '144',\n",
       " '1440',\n",
       " '1444',\n",
       " '1446',\n",
       " '1449-bp',\n",
       " '14490',\n",
       " '145',\n",
       " '145-149',\n",
       " '145-150',\n",
       " '145-base',\n",
       " '145-trisphosphate',\n",
       " '1452',\n",
       " '14579',\n",
       " '145k',\n",
       " '146',\n",
       " '146-amino',\n",
       " '146-kb',\n",
       " '14606-14611',\n",
       " '1462-amino-acid',\n",
       " '1467',\n",
       " '147',\n",
       " '147-kda',\n",
       " '1473',\n",
       " '1474-basepair',\n",
       " '148',\n",
       " '148-amino',\n",
       " '1482',\n",
       " '1485',\n",
       " '149',\n",
       " '1490',\n",
       " '1491-base-pair',\n",
       " '1494',\n",
       " '1495',\n",
       " '14alpha-lanosterol',\n",
       " '14c-glcnac',\n",
       " '14c-labeled',\n",
       " '14cacetic',\n",
       " '14cdehydroepiandrosterone',\n",
       " '14cl-glutamate',\n",
       " '14cpregnenolone',\n",
       " '14k',\n",
       " '14q13-q21',\n",
       " '14q243',\n",
       " '14q31',\n",
       " '14q32',\n",
       " '14q321',\n",
       " '14q323',\n",
       " '14s',\n",
       " '14th',\n",
       " '15',\n",
       " '15-18',\n",
       " '15-20',\n",
       " '15-amino',\n",
       " '15-bisphosphate',\n",
       " '15-carbon',\n",
       " '15-fold',\n",
       " '15-kb',\n",
       " '15-kbp',\n",
       " '15-kda',\n",
       " '15-kilobase',\n",
       " '150',\n",
       " '150-kb',\n",
       " '150-kd',\n",
       " '150-kda',\n",
       " '1500',\n",
       " '15022-15028',\n",
       " '1505',\n",
       " '150519',\n",
       " '150613',\n",
       " '150725',\n",
       " '1509',\n",
       " '151',\n",
       " '151-aa',\n",
       " '1513',\n",
       " '15137-15143',\n",
       " '1514',\n",
       " '15145',\n",
       " '1518',\n",
       " '152',\n",
       " '152-1',\n",
       " '152-156',\n",
       " '152-368',\n",
       " '15233',\n",
       " '1527',\n",
       " '153',\n",
       " '1531974',\n",
       " '1532',\n",
       " '15344',\n",
       " '1537',\n",
       " '1538',\n",
       " '154',\n",
       " '154-14',\n",
       " '154-157',\n",
       " '154-amino',\n",
       " '1542-base-pair',\n",
       " '154478',\n",
       " '1546',\n",
       " '155',\n",
       " '155-fold',\n",
       " '15500',\n",
       " '15500-16800',\n",
       " '155000',\n",
       " '1551335',\n",
       " '156',\n",
       " '156-fold',\n",
       " '1561',\n",
       " '1564905-base-pair',\n",
       " '15650',\n",
       " '1567-amino-acid',\n",
       " '1568',\n",
       " '1569',\n",
       " '15697-nucleotide',\n",
       " '157',\n",
       " '15700',\n",
       " '15719',\n",
       " '1575',\n",
       " '1575-bp',\n",
       " '158',\n",
       " '1580',\n",
       " '159',\n",
       " '1590',\n",
       " '159443-bp',\n",
       " '1596',\n",
       " '1599',\n",
       " '15n-labelling',\n",
       " '15q212',\n",
       " '15q221-2232',\n",
       " '15q24',\n",
       " '15q25',\n",
       " '15th',\n",
       " '16',\n",
       " '16-18',\n",
       " '16-19',\n",
       " '16-30-fold',\n",
       " '16-33',\n",
       " '16-aa',\n",
       " '16-amino',\n",
       " ...]"
      ]
     },
     "execution_count": 341,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import numpy as np \n",
    "import pandas as pd\n",
    "\n",
    "text_df1 = pd.read_csv(\"./trg_copy.csv\")\n",
    "words = []\n",
    "\n",
    "for i in range(0, text_df1.shape[0]):\n",
    "    abstract = text_df1[\"abstract\"][i]\n",
    "    words_abstract = abstract.split(\" \") # Split when there is a space\n",
    "    words_abstract = [x.strip() for x in words_abstract]\n",
    "    words_abstract = [ x for x in words_abstract if x != \"\"]\n",
    "    \n",
    "    words.extend(words_abstract)\n",
    "\n",
    "#words = list(np.unique(np.sort(words)))    \n",
    "words = list(np.sort(np.unique(words)))\n",
    "words\n",
    "#text_df1[\"abstract\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 343,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>'</th>\n",
       "      <th>'87</th>\n",
       "      <th>'a</th>\n",
       "      <th>'a'</th>\n",
       "      <th>'addiction'</th>\n",
       "      <th>'ancient'</th>\n",
       "      <th>'b'</th>\n",
       "      <th>'bird</th>\n",
       "      <th>'birds</th>\n",
       "      <th>'c3-type'</th>\n",
       "      <th>...</th>\n",
       "      <th>zymogen</th>\n",
       "      <th>zymogen-type</th>\n",
       "      <th>zymogens</th>\n",
       "      <th>zymogram</th>\n",
       "      <th>zymography</th>\n",
       "      <th>zymomonas</th>\n",
       "      <th>zzA</th>\n",
       "      <th>zzB</th>\n",
       "      <th>zzE</th>\n",
       "      <th>zzV</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  32245 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   '  '87  'a  'a'  'addiction'  'ancient'  'b'  'bird  'birds  'c3-type'  \\\n",
       "0  0    0   0    0            1          0    0      0       0          0   \n",
       "1  0    0   0    0            0          0    0      0       0          0   \n",
       "2  0    0   0    0            0          0    0      0       0          0   \n",
       "3  0    0   0    0            0          0    0      0       0          0   \n",
       "4  0    0   0    0            0          0    0      0       0          0   \n",
       "\n",
       "   ...  zymogen  zymogen-type  zymogens  zymogram  zymography  zymomonas  zzA  \\\n",
       "0  ...        0             0         0         0           0          0    0   \n",
       "1  ...        0             0         0         0           0          0    0   \n",
       "2  ...        0             0         0         0           0          0    0   \n",
       "3  ...        0             0         0         0           0          0    0   \n",
       "4  ...        0             0         0         0           0          0    0   \n",
       "\n",
       "   zzB  zzE  zzV  \n",
       "0    1    0    0  \n",
       "1    0    0    0  \n",
       "2    0    0    0  \n",
       "3    0    0    0  \n",
       "4    0    0    0  \n",
       "\n",
       "[5 rows x 32245 columns]"
      ]
     },
     "execution_count": 343,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "text_boolean_df = pd.DataFrame( np.zeros((text_df1.shape[0], len(words) + 4), dtype = int) , columns = words + [\"zzA\", \"zzB\", \"zzE\", \"zzV\"])\n",
    "#text_boolean_df[\"class\"] = text_df1[\"class\"]\n",
    "for index in range(0, 1):\n",
    "    classValue = text_df1[\"class\"][index]\n",
    "    if classValue == \"A\":\n",
    "        text_boolean_df[\"zzA\"][index] = 1\n",
    "    elif classValue == \"B\":\n",
    "        text_boolean_df[\"zzB\"][index] = 1\n",
    "    elif classValue == \"E\":\n",
    "        text_boolean_df[\"zzE\"][index] = 1\n",
    "    else:\n",
    "        text_boolean_df[\"zzV\"][index] = 1\n",
    "    \n",
    "#for i in range(0, text_df1.shape[0]):\n",
    "for i in range(0, 1):    \n",
    "    abstract = text_df1[\"abstract\"][i]\n",
    "    words_abstract = abstract.split(\" \") # Split when there is a space\n",
    "    words_abstract = [x.strip() for x in words_abstract]\n",
    "    words_abstract = [ x for x in words_abstract if x != \"\"]\n",
    "    #print(\"words_abstract \", words_abstract)\n",
    "    for j in range(0, len(words_abstract)):\n",
    "        word = words_abstract[j]\n",
    "        #text_boolean_df[word][i] = 1\n",
    "        text_boolean_df.loc[i, word] = 1\n",
    "text_boolean_df.head()\n",
    "#text_boolean_df.to_csv(\"./test_123.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 323,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>'</th>\n",
       "      <th>'87</th>\n",
       "      <th>'a</th>\n",
       "      <th>'a'</th>\n",
       "      <th>'addiction'</th>\n",
       "      <th>'ancient'</th>\n",
       "      <th>'b'</th>\n",
       "      <th>'bird</th>\n",
       "      <th>'birds</th>\n",
       "      <th>'c3-type'</th>\n",
       "      <th>...</th>\n",
       "      <th>zygomycetes</th>\n",
       "      <th>zygotic</th>\n",
       "      <th>zymogen</th>\n",
       "      <th>zymogen-type</th>\n",
       "      <th>zymogens</th>\n",
       "      <th>zymogram</th>\n",
       "      <th>zymography</th>\n",
       "      <th>zymomonas</th>\n",
       "      <th>test</th>\n",
       "      <th>test</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3995</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3996</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3997</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3998</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3999</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4000 rows  32243 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      '  '87  'a  'a'  'addiction'  'ancient'  'b'  'bird  'birds  'c3-type'  \\\n",
       "0     0    0   0    0            1          0    0      0       0          0   \n",
       "1     0    0   0    0            0          0    0      0       0          0   \n",
       "2     0    0   0    0            0          0    0      0       0          0   \n",
       "3     0    0   0    0            0          0    0      0       0          0   \n",
       "4     0    0   0    0            0          0    0      0       0          0   \n",
       "...  ..  ...  ..  ...          ...        ...  ...    ...     ...        ...   \n",
       "3995  0    0   0    0            0          0    0      0       0          0   \n",
       "3996  0    0   0    0            0          0    0      0       0          0   \n",
       "3997  0    0   0    0            0          0    0      0       0          0   \n",
       "3998  0    0   0    0            0          0    0      0       0          0   \n",
       "3999  0    0   0    0            0          0    0      0       0          0   \n",
       "\n",
       "      ...  zygomycetes  zygotic  zymogen  zymogen-type  zymogens  zymogram  \\\n",
       "0     ...            0        0        0             0         0         0   \n",
       "1     ...            0        0        0             0         0         0   \n",
       "2     ...            0        0        0             0         0         0   \n",
       "3     ...            0        0        0             0         0         0   \n",
       "4     ...            0        0        0             0         0         0   \n",
       "...   ...          ...      ...      ...           ...       ...       ...   \n",
       "3995  ...            0        0        0             0         0         0   \n",
       "3996  ...            0        0        0             0         0         0   \n",
       "3997  ...            0        0        0             0         0         0   \n",
       "3998  ...            0        0        0             0         0         0   \n",
       "3999  ...            0        0        0             0         0         0   \n",
       "\n",
       "      zymography  zymomonas  test  test  \n",
       "0              0          0     0     0  \n",
       "1              0          0     0     0  \n",
       "2              0          0     0     0  \n",
       "3              0          0     0     0  \n",
       "4              0          0     0     0  \n",
       "...          ...        ...   ...   ...  \n",
       "3995           0          0     0     0  \n",
       "3996           0          0     0     0  \n",
       "3997           0          0     0     0  \n",
       "3998           0          0     0     0  \n",
       "3999           0          0     0     0  \n",
       "\n",
       "[4000 rows x 32243 columns]"
      ]
     },
     "execution_count": 323,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#text_boolean_df.loc[i, \"addiction\"] = 1\n",
    "#text_boolean_df.head()\n",
    "#text_boolean_df[\"addiction\"][0] = 1\n",
    "#text_boolean_df.head()\n",
    "#text_boolean_df[\"addiction\"] = 1\n",
    "#text_boolean_df[\"addiction\"]\n",
    "#word = words[4]\n",
    "#print(\"word\", word)\n",
    "#text_boolean_df[word][1] = 1\n",
    "#text_boolean_df.head()\n",
    "\n",
    "#data123 = np.zeros((2, 3), dtype = int)\n",
    "words.append(\"test\")\n",
    "data123 = np.zeros((text_df1.shape[0], len(words)), dtype = int)\n",
    "df_1 = pd.DataFrame(data123, columns = words)\n",
    "\n",
    "\n",
    "df_1.loc[0,\"'addiction'\"] = 1\n",
    "df_1\n",
    "\n",
    "#df = pd.DataFrame(data123, columns = words[])\n",
    "#df[\"addition\"][0] = 1\n",
    "#df.at[1, \"addition\"] = 1\n",
    "#df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 454,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 454,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fruits = ['apple', 'banana', 'cherry']\n",
    "\n",
    "found = not (\"cherry1\" in fruits)\n",
    "found\n",
    "#type(found == )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
